### PR TITLE
Do not inherit 'https-client-auth' property for the management interface

### LINKED ISSUE #30094: Do not inherit 'https-client-auth' property for the management interface
### Description

When options for the management interface are not set, they're inherited from the main server. As the management interface now serves only health checks and metrics, it might be reasonable not to require client auth even for the management interface when it is required for the main server.

When users want to assess the mTLS even for the management interface, they'd need to explicitly set the `https-management-client-auth` property.

Relevant issue: https://github.com/keycloak/keycloak/issues/29578

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by mabartos**
Closes #30094

### CODE REVIEW

File: docs/guides/server/management-interface.adoc [hunk 1]
```diff
@@ -32,8 +32,11 @@ if you set the CLI option `--http-relative-path=/auth`, these endpoints are acce
 When the TLS is set for the default {project_name} server, the management interface will be accessible through HTTPS as well.
 The management interface can run only either on HTTP or HTTPS, not both as for the main server.
 
-Specific {project_name} management interface options with the prefix `https-management-*` were provided for setting different TLS parameters for the management HTTP server. Their function is similar to their counterparts for the main HTTP server, for details see <@links.server id="enabletls" />.
-When these options are not explicitly set, the TLS parameters are inherited from the default HTTP server.
+Specific {project_name} management interface options with the prefix `https-management-*` were provided for setting different TLS parameters for the management HTTP server.
+Their function is similar to their counterparts for the main HTTP server, for details see <@links.server id="enabletls" />.
+When these options are not explicitly set, the TLS parameters are inherited from the default HTTP server (except `https-management-client-auth`).
+
+To assess mTLS for the management interface, you need to explicitly set the `https-management-client-auth` option.
```
Comment (shawkins) [inline docs/guides/server/management-interface.adoc:39]:
```suggestion
To enable mTLS for the management interface, you need to explicitly set the `https-management-client-auth` option.
```

File: docs/guides/server/management-interface.adoc [hunk 2]
```diff
@@ -32,8 +32,11 @@ if you set the CLI option `--http-relative-path=/auth`, these endpoints are acce
 When the TLS is set for the default {project_name} server, the management interface will be accessible through HTTPS as well.
 The management interface can run only either on HTTP or HTTPS, not both as for the main server.
 
-Specific {project_name} management interface options with the prefix `https-management-*` were provided for setting different TLS parameters for the management HTTP server. Their function is similar to their counterparts for the main HTTP server, for details see <@links.server id="enabletls" />.
-When these options are not explicitly set, the TLS parameters are inherited from the default HTTP server.
+Specific {project_name} management interface options with the prefix `https-management-*` were provided for setting different TLS parameters for the management HTTP server.
```
Comment (shawkins) [inline docs/guides/server/management-interface.adoc:35]:
```suggestion
Specific {project_name} management interface options with the prefix `https-management-*` are provided for setting different TLS parameters for the management HTTP server.
```

**Review by shawkins — DISMISSED** [review]
Looks good, just a couple of minor suggestions.

**Comment by mabartos** [comment]
@shawkins Thanks, updated.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@mabartos Thank you for the PR. I was thinking about it and I'd probably keep it consistent with other TLS options. It might be confusing that one of them is behaving differently. I wonder if we should rather add some note to [mTLS docs](https://www.keycloak.org/server/enabletls#_enabling_mutual_tls) to highlight it also by default affects the management port. Alternatively, even a generic warning that all options affect the management port? WDYT?

**Comment by mabartos** [comment]
@vmuzikar Thanks for the comment. Even though it makes sense to me not to inherit the property for the management interface, it'd be better to ensure consistency, as you've mentioned. The situation described in the issue happens only when the mTLS is being set up, so it's ok to provide only some notes to the mTLS docs.

Closing as different changes will be provided to close the issue.

**Pull request closed without merging**